### PR TITLE
Abhi/caching

### PR DESCRIPTION
**Pull request by pretentious7**
adds caching for the docker-compose files, for now only S3 is supported; eventually plan to add local caching support for compose.

### CODE REVIEW

File: drone/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,24 @@
-FROM rust:buster as build
+FROM rust:latest as build
```
Comment (paulgb) [inline drone/Dockerfile:1]: Why do this in the drone but not controller?

Comment (pretentious7) [reply drone/Dockerfile:1]: oops oversight

**Comment by paulgb** [comment]
Have you managed to get this to work in `docker compose`?

I set the environment variables, ran `docker compose build`, then modified a file by adding a newline and ran again, and the build still took 5 mins. Can we speed that up?

**Comment by paulgb** [comment]
D'oh deleted a comment because I was using the wrong vars.

**Comment by paulgb** [comment]
Cool, confirmed that this speeds up the docker compose by 5x. Nicely done!

**Review by paulgb — APPROVED** [review]

**Pull request merged**